docker-storage-setup.sh

```diff
@@ -111,12 +115,45 @@ get_deferred_deletion_string() {
   [ -z "$major" ] && return 0
   [ -z "$minor" ] && return 0
 
-  # docker 1.9 onwards supports deferred device deletion. Enable it.
-  if [ $major -gt 1 ] || ([ $major -eq 1 ] && [ $minor -ge 9 ]);then
-    echo "--storage-opt dm.use_deferred_deletion=true"
+  if should_enable_deferred_deletion $major $minor; then
+     echo "--storage-opt dm.use_deferred_deletion=true"
   fi
 }
 
+should_enable_deferred_deletion() {
+   # docker 1.9 onwards supports deferred device deletion. Enable it.
+   local major=$1
+   local minor=$2
+   if [ $major -lt 1 ] || ([ $major -eq 1 ] && [ $minor -lt 9 ]);then
+      return 1
+   fi
+   if platform_supports_deferred_deletion; then
+      return 0
+   fi
+   return 1
+}
+
+platform_supports_deferred_deletion() {
+        local deferred_deletion_supported=1
+        trap cleanup_pipes EXIT
+        mkfifo $PIPE1
+        mkfifo $PIPE2
+        mount -o bind $TEMPDIR $TEMPDIR
+        unshare -m ./dss-child-read-write.sh $PIPE1 $PIPE2 &
```

Comment:
I think this patch is almost there. One minor nit. I don't think we should hardcode the path of the dss-child-read-write.sh. May be let it be found in $PATH. So just call it directly witout `./`

unshare -m dss-child-read-write.sh $PIPE1 $PIPE2 &.

Another concern is what if this program is not installed or missing. We probably want to capture error code. Otherwise next "read" call will block infinitely.